scripts/template_matcher.py

```diff
@@ -59,15 +84,58 @@ def _compute_prediction(self, k, img, kp, des):
         kp: keypoints from scene image,   des: descriptors from scene image
         """
 
-        # TODO: find corresponding points in the input image and the templae image
-        #       put keypoints from template image in template_pts
-        #       put corresponding keypoints from input image in img_pts
-
-
-        #TODO: change img to img_T once you do the homography transform
-        visual_diff = compare_images(img, self.signs[k])
+        # Get list of matches between keypoints based on destriptors
+        matches = self.matcher.knnMatch(self.descs[k],des,k=2)
+        # Convert into point form
+        matches = [(m.queryIdx, m.trainIdx) for m, n in matches]
```

Comment:
It's here where using a technique like Lowe's Ratio Test helps in filtering out true / false positive matches.

Here's an example of Paul using ratio_threshold to implement Lowe's Ratio Test.
https://github.com/paulruvolo/comprobo17/blob/master/computer_vision_examples/scripts/match_keypoints.py#L52

See Figure 11 too in Lowe's Original SIFT paper. https://www.cs.ubc.ca/~lowe/papers/ijcv04.pdf